### DISCUSSION TITLE
switchmode 6 not sending toggle payload

### DISCUSSION THREAD

**Discussion by GsynthChris** [arendst/Tasmota#12754] [Q&A] [answered]
Hello,

Tasmota 9.5.0 (tasmota-ir.bin)

First, this is a repeat project for me. I made this years ago but recently my esp8266 board went bad and I am trying to get myself set back up and I didn't have a backup of the config from the old one. 

I made a switch plate for in my bedroom with 4 capacitive touch buttons set up as switch1-4 and it also has an infrared LED on it for tuning on my TV and adjusting volume.  Before my old board went bad I believe it was on Tasmota-ir 8.1.

The IR part works fine, no problems there.

The switches are giving me a hard time. I had it set up before so a short press would send the `TOGGLE` payload and a long press would send `HOLD`.  I am using `switchmode# 6` on all 4 switches. 

Problem 1: I can only get it to send `POWER1`, `POWER2`, etc, if I set up fake relays on unused GPIOs. I swear I didn't have to do that in the past. I know I have ran into this problem before but I can't remember how I fixed it.  Not that its a big deal but I would like to be able to just send an MQTT payload without needing the fake relay set up. I tried `setoption114 1` but it sends a completely different payload than I am used to seeing and I can't figure out how to make use of it. 

Problem 2: I can't get it to send `TOGGLE `with a short press. Instead of sending `TOGGLE`, it alternates sending `ON `or `OFF`. If I do a long press I do get the `HOLD `payload. I tried setting up a rule but I don't think what I tried is working. 
I just need it to send `TOGGLE `when the circuit is closed and nothing when released
The rule I tried is `rule1 on switch1#state=1 do publish cmnd/bedroomremote/POWER1 2 endon`

**Comment by ascillato2** [comment, marked as answer]
Please, address this to the [Tasmota Support Chat](https://discord.gg/Ks2Kzd4). The chat is a better and more dynamic channel for helping you. Thanks.

---------------------------

### Support Information 
See [Docs](https://tasmota.github.io/docs/) for more information.
See [Chat](https://discord.gg/Ks2Kzd4) for more user experience.
See [Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
